### PR TITLE
Populate ConnectivityResult in SA post contingency results

### PR DESCRIPTION
**Pull request by gverger**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Connectivity Results have been added to the API of Post Contingency Results for Security Analysis (issue: https://github.com/powsybl/powsybl-core/issues/2183, PR: https://github.com/powsybl/powsybl-core/pull/2413)
Here we want to populate these results.

**What is the current behavior?** *(You can also link to an open issue here)*
Currently the connectivity results are not computed and a default value is returned.

**What is the new behavior (if this is a feature change)?**
In the PostContingencyResults, we will now have the number of synchronous components created when applying a contingency, and the load and generation lost.

This PR is not meant to deal with connected components since there is more work to be done with HVDC lines.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/LfContingency.java [hunk 1]
```diff
@@ -75,6 +92,10 @@ public int getIndex() {
         return index;
     }
 
+    public int getNbSynchronousComponents() {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/LfContingency.java:95]: I think that here you have to store directly the number of components created by the contingency and add the -1 in that class.

File: pom.xml [hunk 2]
```diff
@@ -47,7 +47,6 @@
         <jafama.version>2.3.2</jafama.version>
         <junit-jupiter.version>5.8.2</junit-jupiter.version>
         <slf4jtoys.version>1.6.3</slf4jtoys.version>
-
```
Comment (annetill) [inline pom.xml:50]: Can you leave the space please?

File: src/main/java/com/powsybl/openloadflow/dc/equations/DcEquationSystemUpdater.java [hunk 3]
```diff
@@ -45,6 +45,7 @@ public void onDisableChange(LfElement element, boolean disabled) {
                 break;
             case BRANCH:
             case HVDC:
+            case SHUNT_COMPENSATOR:
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/dc/equations/DcEquationSystemUpdater.java:48]: @obrix can you remove this fix? We have to open a dedicated PR that shows the bug. Can you open it in parallel and I am going to do a use case. Thanks !

Comment (obrix) [reply src/main/java/com/powsybl/openloadflow/dc/equations/DcEquationSystemUpdater.java]: Okay, I just opened a separate pull request.

File: src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java [hunk 4]
```diff
@@ -231,8 +245,8 @@ private List<OperatorStrategyResult> createOperatorStrategyResults(DcSecurityAna
         try (LfNetworkList lfNetworks = Networks.load(network, dcParameters.getNetworkParameters(), allSwitchesToOpen, allSwitchesToClose, Reporter.NO_OP)) {
             return lfNetworks.getLargest().filter(LfNetwork::isValid)
                     .map(largestNetwork -> runActionSimulations(context, largestNetwork, dcParameters, propagatedContingencies,
-                            operatorStrategies, actionsById, neededActions))
-                    .orElse(Collections.emptyList());
+                                operatorStrategies, actionsById, neededActions)
+                    ).orElse(Collections.emptyList());
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java:249]: @annetill fix this

File: src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java [hunk 5]
```diff
@@ -294,9 +307,39 @@ private List<OperatorStrategyResult> runActionSimulations(DcSecurityAnalysisCont
...
+                                lfContingency.apply(context.getParameters().getLoadFlowParameters().getBalanceType());
+                                // we build the connectivity result linked to this contingency by opportunity.
+                                ConnectivityResult connectivityResult = new ConnectivityResult(lfContingency.getCreatedSynchronousComponentsCount(), 0,
+                                        lfContingency.getDisconnectedLoadActivePower() * PerUnit.SB,
+                                        lfContingency.getDisconnectedGenerationActivePower() * PerUnit.SB,
+                                        lfContingency.getDisconnectedElementIds());
+                                networkState.restore();
+                                return connectivityResult;
+                            })
+                            .orElse(new ConnectivityResult(0, 0, 0, 0, Collections.emptySet()))
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java:338]: We have a API design issue, maybe we should have done this ConnectivityResult optional.

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java:338]: If you have a `PostContingencyResult`, you have a `ConnectivityResult`, no?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java:338]: Bu if you don't have any loss of connectivity for you contingency, you should not provider this object?

Comment (annetill) [reply src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java:338]: No... right...

**Review by geofjamg — APPROVED** [review]

**Review by annetill — APPROVED** [review]

**Pull request merged**